### PR TITLE
feat(rules): create rule directly from transaction detail view

### PR DESCRIPTION
**Pull request by gfr-free**
## Summary
- Adds a "Create rule" action to the transaction detail drawer's Settings section, between "Mark as Recurring" and "Delete" — closes #2960.
- Reuses the existing `RulesController#new` query-param prefill (`name` → `transaction_name`/`like` condition); no new controller logic. The link uses the transaction's merchant name if set, otherwise the transaction/payee name.
- Deliberately minimal prefill: no action is pre-selected, the rule is not auto-applied. The user reviews/edits everything in the existing rule editor modal, then goes through the existing confirm/apply flow to (optionally) run it against history — consistent with how every other rule is created today.

## Test plan
- [x] `bin/rails test test/controllers/transactions_controller_test.rb` — 4 new tests: merchant-name prefill, name fallback without a merchant, hidden for split-child transactions, hidden without edit permission on a shared read-only account.
- [x] `bin/rails test test/models/rule_test.rb test/controllers/rules_controller_test.rb test/i18n_test.rb` — no regressions (existing `RulesController#new` prefill path is unchanged).
- [x] `bin/rubocop -f github -a` — clean.
- [x] `bundle exec erb_lint -a` on the touched views — clean.
- [x] `bin/brakeman --no-pager` — 0 warnings.
- [x] Manually verified in a running instance: opened a transaction, "Create rule" appears between "Mark as Recurring" and "Delete", opens the rule editor modal with the name condition pre-filled from the transaction's merchant.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a **Create Rule** option to the transaction details drawer.
  * Rule creation is prefilled using the transaction’s merchant when available, or its name otherwise.
  * The option is available for editable transactions and hidden for split child transactions or when editing isn’t permitted.
  * The create-rule link refreshes after transaction updates, including validation failures, to reflect current details.

* **Bug Fixes**
  * Invalid merchant identifiers now safely fall back to name-based rule creation.

* **Tests**
  * Added coverage for prefilling, link updates, fallback behavior, and visibility rules.

### CODE REVIEW

**Comment by gfr-free** [comment]
Both findings are valid, fixed in 2d76c68:

- **Merchant condition prefill**: `assign_create_rule_state` now checks for a merchant and, when present, prefills a `transaction_merchant` condition scoped to the merchant's id (via `Family#available_merchants_for`) instead of a `transaction_name` condition built from the merchant's display name — the latter queried `entries.name`, not the merchant, so it could silently fail to match.
- **Stale link after edits**: `assign_create_rule_state` is now called after the `update` reload (success path) and in the validation-failure branch before `render :show`. The Turbo Stream success response also replaces the `create_rule` element so the link reflects the just-edited name/merchant.

Added regression tests for the merchant-id prefill, the update-refreshes-link case, and the failed-validation case.

**Comment by gfr-free** [comment]
Good catch, fixed in ab52ac1: switched both assertions to \`assert_select "a[href=?]"\` (scoped to the \`create_rule\` turbo-stream target, with an explicit negative check for the stale Netflix href) instead of raw-body \`assert_match\`/\`assert_no_match\`, which was comparing against unescaped \`&\` while \`link_to\` renders \`&amp;\`.

While fixing this I also caught that CI had failed on the "update fails validation" test — it used \`entry[date]=nil\`, but \`TransactionsController#entry_params\` strips a blank \`:date\` from the params before calling \`update\`, so that never actually produced a validation failure (200, not 422). Switched to \`entry[name]=""\`, which passes through unmodified and hits \`Entry\`'s \`validates :name, presence: true\`.

File: app/controllers/rules_controller.rb [hunk 1]
```diff
@@ -30,7 +30,14 @@ def new
       resource_type: params[:resource_type] || "transaction",
     )
 
-    if params[:name].present?
+    if params[:merchant_id].present? && (merchant = Current.family.available_merchants_for(Current.user).find_by(id: params[:merchant_id]))
```
Comment (jjmata) [inline app/controllers/rules_controller.rb]: A malformed (non-UUID) `merchant_id` will crash this request instead of falling through to the `elsif params[:name].present?` name-only prefill.

`merchants.id` is a Postgres `uuid` column. `find_by(id: params[:merchant_id])` passes the raw query-string value straight into the WHERE clause, so a non-UUID value (e.g. a hand-edited/stale `?merchant_id=not-a-uuid` link) raises `PG::InvalidTextRepresentation` → `ActiveRecord::StatementInvalid`. There's no `rescue_from` for `StatementInvalid` anywhere in the app (only `ActiveRecord::RecordNotFound` is rescued globally, e.g. `app/controllers/concerns/store_location.rb:9`), so this 500s instead of hitting the fallback branch the code clearly intends here.

Since `merchant_id` becomes a newly user/attacker-reachable query param via this PR's own generated link, worth guarding — e.g. rescue `ActiveRecord::StatementInvalid` around the lookup, or validate the param is a UUID before querying.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by jjmata — COMMENTED** [review]
Reviewed the diff. One correctness finding below — a code path in `RulesController#new` that can 500 on a malformed `merchant_id`. Everything else (merchant-aware condition prefill, refreshing the create-rule link on update/failed-validation, i18n, test coverage) looks correct.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: app/controllers/rules_controller.rb (hunk 1 above)
Comment (gfr-free) [reply app/controllers/rules_controller.rb]: Fixed in afb7c4746 — guarded the `merchant_id` lookup with `UuidFormat.valid?` (the same helper used elsewhere in the app, e.g. `api/v1/base_controller.rb`) before the `find_by`, so a malformed value now falls through to the name-only prefill instead of raising `ActiveRecord::StatementInvalid`. Added a regression test covering this case.
